client/src/pages/JoinPage/index.tsx

```diff
@@ -21,20 +30,73 @@ const JoinPage = ({
   const { socket } = useContext(SocketContext);
   const { stream, setStream } = useContext(DeviceContext);
   const { room_id } = useParams<{ room_id: string }>();
-  const [showNotification, setShowNotification] = useState(false);
+  const [notificationState, setNotificationState] = useState({
+    text: "",
+    isOpen: false,
+  } as NotificationState);
```

Comment:
You can also use the syntax 

```typescript
const [notificationState, setNotificationState] = useState<NotificationState>(...)
```

instead of using `as`

Reply:
Oh right I forgot I'll change it to that thanks!

Reply:
You don't have to make changes to this, since all the `useStates` are not typed, but just for future reference